### ISSUE TITLE
[create-app] the `master` nomenclature branch naming is outdated and hardcoded

### ISSUE THREAD

**Issue by planeiii-te** [backstage/backstage#13408]
## Expected Behavior

Either the name of the default branch (typically `main`) should be read from the environment or requested from the user.

I think reading from the environment offers the best user experience.

## Actual Behavior

The antiquated name is used and required to be changed after template generation.

https://github.com/backstage/backstage/blob/cdc6515e8f055ebe74198cdda4107f35d51259c8/packages/create-app/templates/default-app/package.json.hbs#L20

## Steps to Reproduce

1. `git config --global --set init.defaultBranch main`
2. `npx @backstage/create-app` and follow the prompts

## Context

https://github.com/github/renaming

## Your Environment

- Browser Information: 

- Output of `yarn backstage-cli info`: 

```text

```

**Comment by Rugvip** [comment]
Agreed. I think while at it we should really have `create-app` initialize a new git repo too and commit the resulting app in an initial commit, or maybe two, before and after `yarn install`. Alongside that it should be simple to read out a couple of config values too that we can make available during templating, I'd very much like to avoid more user input.

Is this something you'd like to work on or should we open it up for contributions?

**Comment by leonardomaier** [comment]
I spent a few hours trying some approaches.

First I added the `defaultBranch` context variable on `package.json.hbs`:

```json
"lint": "backstage-cli repo lint --since origin/{{ defaultBranch }}",
```

On `createApp.ts` I've added some additional steps regarding the git repository initialization like @Rugvip mentioned.

```ts
/* Additional steps */

Task.section('Initializing git repository');
// It runs git init && touch README.md
// I'm adding the README.md file since at least one file is needed to be able to commit
// and later on the file is going to be replaced by the one on the template folder
await initGitRepository(tempDir);

Task.section('Creating first commit');
// It runs git add . && git commit -m "Initial commit"
await gitCommit(tempDir, "Initial commit");      

Task.section('Getting default branch name');
// It runs git branch --format="%(refname:short)" to get the newly created branch
// and adds the branch name to answers object (answers.defaultBranch)
await getDefaultBranchName(tempDir, answers);

/* End of additional steps */

Task.section('Preparing files');
await templatingTask(templateDir, tempDir, answers);

Task.section('Moving to final location');
await moveAppTask(tempDir, appDir, answers.name);
```
I'm using the temporary directory to initialize the git repo since we need everything ready before the `templatingTask` function call which is the one that replaces all template variables.

The written code is not even close to the final solution, I'm going to refactor it later on, just wanted to confirm if we're on the same page before moving on. 

Is that close to what you were thinking @Rugvip?

**Comment by Rugvip** [comment]
@leonardomaier yep something like that! You can use `--allow-empty` to avoid the need for the dummy readme. I'd say that's good to go for a PR if you want to open one up, and we can iron out the details there

**Comment by leonardomaier** [comment]
@Rugvip good to hear that, I'll open a PR soon. Could you assign me to the issue?

**Comment by Rugvip** [comment]
This was fixed by #13585, closing
